dev/RadialGradientBrush/RadialGradientBrush.cpp

```diff
@@ -0,0 +1,228 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "Vector.h"
+#include "SharedHelpers.h"
+#include "RadialGradientBrush.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+
+RadialGradientBrush::RadialGradientBrush()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_RadialGradientBrush);
+
+    m_gradientStops = winrt::make<Vector<winrt::GradientStop>>().as<winrt::IObservableVector<winrt::GradientStop>>();
+}
+
+winrt::IObservableVector<winrt::GradientStop> RadialGradientBrush::GradientStops()
+{
+    return m_gradientStops;
+}
+
+void RadialGradientBrush::OnConnected()
+{
+    // XCBB will use fallback rendering in design v1 mode, so do not create a CompositionBrush.
+    if (SharedHelpers::IsInDesignMode()) { return; }
+
+    m_isConnected = true;
+
+    EnsureCompositionBrush();
+
+    if (SharedHelpers::IsCompositionRadialGradientBrushAvailable())
+    {
+        // If CompositionRadialGradientBrush will be used then listen for changes to gradient stops so the composition brush can be updated.
+        m_gradientStopsChangedToken = m_gradientStops.VectorChanged({
+            [this](winrt::IObservableVector<winrt::GradientStop> const& sender, winrt::IVectorChangedEventArgs const& args)
+            {
+                if (m_brush)
+                {
+                    UpdateCompositionGradientStops();
+                }
+            }
```

Comment:
Should this be its own method? if not, should OnFallbackColorChanged be a lambda?

Reply:
Precedent seems to be that DP changes get their own methods, but VectorChanged is a mix. I'll switch to methods for both.